lib/routes/developer/activate.js

```diff
@@ -0,0 +1,41 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const auth = require('../../auth');
+const db = require('../../db');
+const hex = require('buf').to.hex;
+
+/*jshint camelcase: false*/
+module.exports = {
+  auth: {
+    strategy: auth.AUTH_STRATEGY,
+    scope: [auth.SCOPE_CLIENT_MANAGEMENT]
+  },
+  handler: function activateRegistration(req, reply) {
+    var email = req.auth.credentials.email;
+
+    return db.getDeveloper(email)
+      .then(
+        function(developer) {
+          if (developer) {
+            developer.developerId = hex(developer.developerId);
+            return reply(developer).code(200);
```

Comment:
Also, I'm confused by this return here. It will return a `reply` object, so the promise chain will continue, right? Meaning the db will get queried and mutated after the response of 200 was sent.